datamodel/src/main/java/com/hartwig/serve/datamodel/ActionableEventComparator.java

```diff
@@ -23,9 +23,37 @@ public int compare(@NotNull ActionableEvent event1, @NotNull ActionableEvent eve
             return directionCompare;
         }
 
-        int treatmentCompare = event1.treatment().name().compareTo(event2.treatment().name());
-        if (treatmentCompare != 0) {
-            return treatmentCompare;
+        ClinicalTrial clinicalTrialEvent1 = null;
+        ClinicalTrial clinicalTrialEvent2 = null;
+        Treatment treatmentEvent1 = null;
+        Treatment treatmentEvent2 = null;
+
+        if (event1.intervention() instanceof ClinicalTrial) {
+            clinicalTrialEvent1 = (ClinicalTrial) event1.intervention();
+        }
+
+        if (event2.intervention() instanceof ClinicalTrial) {
+            clinicalTrialEvent2 = (ClinicalTrial) event2.intervention();
+        }
+        if (event1.intervention() instanceof Treatment) {
+            treatmentEvent1 = (Treatment) event1.intervention();
+        }
+        if (event2.intervention() instanceof Treatment) {
+            treatmentEvent2 = (Treatment) event2.intervention();
+        }
+
+        if (clinicalTrialEvent1 != null && clinicalTrialEvent2 != null) {
+            int trialCompare = clinicalTrialEvent1.studyNctId().compareTo(clinicalTrialEvent2.studyNctId());
+            if (trialCompare != 0) {
+                return trialCompare;
+            }
+        }
+
+        if (treatmentEvent1 != null && treatmentEvent2 != null) {
+            int treatmentCompare = treatmentEvent1.name().compareTo(treatmentEvent2.name());
+            if (treatmentCompare != 0) {
+                return treatmentCompare;
+            }
```

Comment:
Quite minor but you could define an ordering between trials and treatments. E.g. should trials always be before treatments?

Reply:
This is indeed a good point, but didn't see how this is possible

Reply:
You can for instance do `if clinicalTrialEvent1 != null && clinicalTrialEvent2 == null` to check whether one entry is a trial and the other one isn't

Reply:
I have implement the order to clinical trial --> treatment

Reply:
Clear! I have improved the test a bit and done some cleanup on the code, see latest commit.